### ISSUE TITLE
Lazy visualizations 

### ISSUE THREAD

**Issue by sylwiabr** [enso-org/ide#982]
### Summary

Since we are starting to work with dataframes we need to implement laziness in visualizations. The data in the dataframes are too big to be resend and displayed after each action.

**Comment by MichaelMauderer** [comment]
Is the preProcessor code going to be something hard-coded in the visualization, or is this going to be some form of input from the IDE (e.g., a new node input for visualizations that can be controlled by the user)?

**Comment by farmaazon** [comment]
Each time we attach a visualization we send to Engine a lambda preprocessing data returned from node. This is the "expression" field of [VisualizationConfiguration](https://dev.enso.org/docs/enso/language-server/protocol-language-server.html#visualisationconfiguration). We can also modify it in already attached vis.

**Comment by MichaelMauderer** [comment]
Do we have some comprehensive documentation about what the inputs for the preprocessor expression are and where the outputs are going? If not, we should add those somewhere.  I'm thinking again about how the user can pass settings to the visualization.

**Comment by farmaazon** [comment]
No, we haven't. Probably should go [here](https://dev.enso.org/docs/enso/language-server/protocol-language-server.html#visualisationconfiguration) and/or [here](dev.enso.org/docs/ide/product/visualizations.html). I add a note to some issue for making good docs for that.

**Comment by MichaelMauderer** [comment]
That would be great! A big picture overview of the envisioned pipeline would be super useful.
